vision_model/trainer_callback_interface.py

```diff
@@ -0,0 +1,51 @@
+import abc
+import typing
+import torch
+
+
+class TrainerCallbackInterface(abc.ABC):
+    """Interfaces a Trainer provides for callback methods"""
+    @property
+    @abc.abstractmethod
+    def train_dataloader(self) -> torch.utils.data.DataLoader:
+        raise NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def val_dataloader(self) -> typing.Optional[torch.utils.data.DataLoader]:
+        raise NotImplementedError
+
+    @property
```

Comment:
Any reason to include the num_epochs here, even with it being part of the training config?

Reply:
Trainer can implement any kind of training config. To avoid slowing down trainer development, I'm not including trainer config interface in this package.

Reply:
yeah, but do we need num_epochs in the trainer as property

Reply:
On second thought, model plugins don't require this property. I've simplified the interfaces and remove this property